docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,524 @@ overview page.
+
+Check `here <https://docs.fluentbit.io/manual/pipeline/outputs>`__
+for the official list of supported outputs.
+
+Loki
+""""
+
+Loki has been chosen as the distributor, ingester & querier because, like
+Fluent Bit, it has a really small impact on resources and is very cost
+effective regarding storage needs.
+
+It also uses the LogQL syntax for queries, which is pretty close to what
+we already have with Prometheus and PromQL, so it eases the integration
+in our tools.
+
+Rejected Design Choices
+-----------------------
+
+Logging Operator
+~~~~~~~~~~~~~~~~
+
+`Logging Operator`_ seemed to be a good candidate for the implementation
+we choose, offering the ability to deploy and configure Fluent Bit and Fluentd,
+but Fluentd is not optional and seems to have a central place as most of the
+parsing/filtering is done by this one, which means a bigger footprint on the
+hardware resources.
+
+.. _Logging Operator: https://github.com/banzaicloud/logging-operator
+
+Promtail + Loki
+~~~~~~~~~~~~~~~
+
+Promtail has been rejected because, even if it consumes very few resources,
+it can only integrate with Loki and we need something more versatile, with
+the ability to interact with different distributors.
+
+Fluentd + Loki
+~~~~~~~~~~~~~~
+
+This architecture has been rejected because it means we need 1 Fluentd instance
+per node, which increases a lot the resources consumption compared to other
+solutions.
+
+Fluent Bit + Fluentd + Loki
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We have considered this architecture as the Fluent Bit + Fluentd couple seems
+to be a standard in the industry, but we didn't find any reason of keeping
+Fluentd, apart for its large panel of plugins which we don't really need.
+Fluent Bit alone seems to be sufficient for what we want to achieve and adding
+an extra Fluentd means more resources consumption and add unnecessary
+complexity in the log centralization stack.
+
+Logstash + Elasticsearch
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This architecture is probably the most common one, but it has not been taken
+into consideration because we want to focus on having the minimum resources
+consumption and these components can really hog RAM & CPU. Beats could be used
+as the log collector to reduce the impact of Logstash, but Elasticsearch still
+consumes a lot of resources.
+Even if this solution offers a lot of powerful functionalities
+(e.g. distributed storage, full-text indexing), we don't really need them and
+want to focus on the smallest hardware footprint.
+
+Implementation Details
+----------------------
+
+Deployment
+~~~~~~~~~~
+
+All the components will be deployed using Kubernetes manifests through Salt
+inside a ``metalk8s-logging`` Namespace.
+
+Fluent Bit
+""""""""""
+
+Fluent Bit will be deployed as a DaemonSet, because we need one collector on
+each node to be able to collect logs from both the Kubernetes platform, the
+applications that run on top of it and all the system daemons running alongside
+(e.g. Salt minion).
```

Comment:
Yes, exempli gratia is not intented to be exhaustive, but maybe since Kubelet is a really important component, it could make sens to add it here. I'll do it.